cEP-0022.md

```diff
@@ -0,0 +1,162 @@
+   so will lead to dropping of most **optional** bear settings, thus will lead to
+   coala not detecting many inconsistencies, and the developers of the project will
+   have to do the work of learning about the bears and their settings and then
+   modifying the `.coafile` manually, while not able to guess the correct
+   **necessary** bear settings will lead to dropping of bears themselves.
+
+**The following steps mainly describe the brute force way**
+
+2. Certain thresholds may be defined for an org based on several factors
+   for the number of inconsistencies detected to be considered mistakes however
+   these shall be stored for later as the first and foremost goal of this project
+   is to generate completely *"green"* config files.
+
+3. coala will be ran over and over again for the entire project with all
+   instances of bears with all the combinations of settings running in
+   parallel, as soon as instances of a particular setting is done and we get
+   a value for the setting for which no errors were generated, it goes directly
+   into the `.coafile`, if no value for an optional setting matches, the setting will
+   have to be dropped,  while if no necessary setting is found out to be *green*,
+   the entire bear will have to be dropped, keeping at the back (of our heads) that,
+   certain settings producing number of errors within the threshold as described in
+   step 2 are being saved for later.
+
+4. Once we have collected this data and using the metadata already being
+   collected from `coala-quickstart` we will generate our first and foremost complete
+   *green* config file.
+
+5. The additional data generated for erraneous settings within a theshold can
+   be reported later to the project administrators in any form and wait for their
+   decision whether these are actual codestyles or mistakes. In case of mistakes,
+   `coala-quickstart` should have the ability to add these rules automatically
+   to the config file. In case these are actual codestyles, `coala-quickstart`
+   should store this data for use in the next phase.
+
+# Improving upon the Brute Force
+
+Premature optimization should not be performed on anything
+although some of the limitations of the brute force method are clear even
+at the beginning. We try to rush through the brute force as quick as possible
+and in the mean time keep on thinking of enhancing this in ways to detect more
+specific bear settings and reducing the total run time of itself.
+
+## The *"lite-mode"*
+
+From what is evident, the brute-force is going to take a huge amount of time
+to run especially on large projects. For this the `coala-quickstart` will
+have a `lite-mode`. We assume that the lowest level in which there can be
+variations will be file types as a starting point, we try to guess the
+settings not from the entire project, but for each file type, we grab a set
+of files at random and run coala and again look for *green* settings, if we
```

Comment:
Maybe we could come up with something cleverer than random; most touched files (according to git? I think we already have this functionality in gitmate somewhere), largest files, most recently modified files? Probably overengineering something. Might even result in worse results. Just wanted to note this somewhere.